src/BizHawk.Emulation.Cores/Arcades/MAME/MAME.cs

```diff
@@ -129,15 +131,50 @@ private void StartMAME(List<IRomAsset> roms)
 
 			var gameName = _gameFileName.Split('.')[0];
 
+			static byte[] MakeRomData(IRomAsset rom)
+			{
+				if (rom.Extension.ToLowerInvariant() is ".zip")
+				{
+					// if this is deflate, unzip the zip, and rezip it without compression
+					// this is to get around some zlib bug?
+					using var ret = new MemoryStream();
+					ret.Write(rom.FileData, 0, rom.FileData.Length);
+					using var zip = new ZipArchive(ret, ZipArchiveMode.Update, leaveOpen: true);
+					foreach (var entryName in zip.Entries.Select(e => e.FullName).ToList())
+					{
+						try // TODO: this is a bad way to detect deflate
+						{
+							var oldEntry = zip.GetEntry(entryName)!;
+							using var oldEntryStream = oldEntry.Open(); // if this isn't deflate, this throws
+							var contents = oldEntryStream.ReadAllBytes();
+							oldEntryStream.Dispose();
+							oldEntry.Delete();
+							var newEntry = zip.CreateEntry(entryName, CompressionLevel.NoCompression);
+							using var newEntryStream = newEntry.Open();
+							newEntryStream.Write(contents, 0, contents.Length);
+						}
+						catch
+						{
+							// ignored
+						}
+					}
+
+					zip.Dispose(); // this actually modifies the MemoryStream

```

Comment:
Style nit --  this would be better as a `using` block

Reply:
Dispose needs to be called before ret.ToArray() is done. In hindsight, I would suppose another pair of parens would work for that.

Reply:
This is a using block, as opposed to a using declaration:

```C#
using (var zip = new ZipArchive(ret, ZipArchiveMode.Update, leaveOpen: true))
{
  // ...
}
```